Makefile

```diff
@@ -115,9 +115,12 @@ build/%.asc:
 
 test: test-unit ## Run the tests
 
-test-linter:  ## Run golangci-lint for the project
+test-linter:  ## Run golangci-lint with the enabled checks (used in the CI)
 	./hack/go-linter.sh
```

Comment:
Wondering if we should remove this make target and just call the linter script from sanity test script? We already have a bunch of make targets here and we should encourage devs to run the entire sanity test suite before pushing and triggering the more resource intensive e2e tests in CI.

Reply:
IHMO: as a dev, I'd like to call this one and see just the linter errors caused because of my PR changes. I understand that the other (dev) one should be something temporary and I do not see in the practical daily base we using that too much.  

So, because of this I'd like to have a target for that.

Reply:
WDYT @jmrodri @fabianvf @estroz @jmccormick2001 @hasbro17 ?

Reply:
Following an example to illustrate

<img width="1419" alt="Screenshot 2019-11-22 at 14 26 20" src="https://user-images.githubusercontent.com/7708031/69433802-3e208300-0d34-11ea-88eb-f7a62b275f9f.png">
:

Reply:
Just to clarify in case the motivation I have in mind isn't clear. I'm suggesting we fold the CI linter (the one that we expect not to return any lint errors) into the sanity test script.

The idea is that just before we push our commits, we can run `make test-unit test-sanity` and check to make sure both of those pass.

The downside would be that to see lint errors, one would need to run the sanity test that does some other things, which takes longer. But I'm only going to run the CI linters before pushing, just like I only run the sanity tests before pushing. In my mind, the CI linters are sanity checks (i.e. "Make sure I didn't introduce any lint errors").

If this is a separate make target, we will now have to remember to ALSO run this one before pushing our commits and triggering CI.

Reply:
Done.

Reply:
I'm okay with the current change since there is still a `lint:` task. But I would've simply made `test-linter` be a dependency of `test-sanity` and that would've fixed it:
```
test-sanity test/sanity: tidy ... test-linter
     ./hack/sanity-script.sh
```
But the current iteration is okay.